### ISSUE TITLE
Naming a block "constructor" fails (from forum)

### ISSUE THREAD

**Issue by brianharvey** [jmoenig/Snap#3401]
https://forum.snap.berkeley.edu/t/naming-a-new-block-constructor-will-break-the-editor/18059

I just found this very odd bug. Here is how to reproduce it.

1. Create a new block named "constructor" (case sensitive) in any category. It can be a command, reporter, or predicate.
2. The make a block window will not close when you press OK, and the block will not show up on the left.
3. Press cancel on the make a block window
4. Click to another category.
5. Click back to the category the "constructor" was created in. The pallet will be blacked out, and blocks cannot be taken out of it.

This problem is persistent. Saving and loading the project will not fix it. I am currently using 10.0.13 on this website. Is the something known about? Are there any other words to avoid? Thanks!

**Comment by cycomachead** [comment]
Whoa, I assume this is because JavaScript has `constructor` the initialization method when using `class` syntax and the block spec is likely getting put into an object as some key? I haven't looked at a trace yet, LOL

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/constructor

I don't know if there's any other odd cases, but maybe `__proto__` would mess things up too.

**Comment by cycomachead** [comment]
Yeah, it seems like any built-in object property has this trait:

(For this list, you need only the function name, not the `()` or arguments. This is because we put strings in an Object, and JS maps strings to function names the same way. I think there are likely some other terms, but I tried `__proto__` and `toString` and those both had the same issues.

```
__defineGetter__(propertyName, getterFunction)

__defineSetter__(propertyName, setterFunction)

__lookupGetter__(propertyName)

__lookupSetter__(propertyName)

constructor: function()

hasOwnProperty(propertyName)

isPrototypeOf(property)

propertyIsEnumerable(propertyName)

toLocaleString()

toString()

valueOf()
```

**Comment by brianharvey** [comment]
Works in 9.2.17, so it's a new bug, in case that wasn't clear.

**Comment by jmoenig** [comment]
thanks!
Fixed this in 10.1  for all such cases. Good catch.
